src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/repository/BackupRepository.java

```diff
@@ -51,7 +51,7 @@
 
     private static final String CONFIGURATION_BACKUP_PATH = SystemProperties.getConfBackupPath();
     // Criteria for valid backup file names:
-    // 1) cannot start with ".", 2) must contain one or more word characters ([a-zA-Z_0-9]), 3) must end with ".tar"
+    // 1) cannot start with ".", 2) must contain one or more word characters ([a-zA-Z_0-9.-]), 3) must end with ".tar"
     private static final String BACKUP_FILENAME_PATTERN = "^(?!\\.)[\\w\\.\\-]+\\.tar$";
```

Comment:
I would leave the validation in the repository layer for security reasons. In case the repository layer is accessed from multiple classes, each class must implement its own validation. When the validation is implemented on the repository layer, the validation is always guaranteed and it's done using the same rules. Since the filename comes from the client, it's extremely important that it's validated properly before it's used.

Reply:
Happened to see this in the github stream and got curious, so had a look. My two cents:
- validation should in general follow the layer design @carohauta linked - one reason is, that we should follow _principle of least astonishment_ and not confuse maintainers. 
- if there is a special case, where we are concerned of security, and are afraid that someone will add a service method that forgets the validation, I think that is a valid reason to have the validation **also** on repository layer. But I think it should not be missing from service layer. I was happy to see it is not!
- Of course, if we are scared of someone adding a new service method and forgetting validation, we should be scared of someone adding a new repository method and forgetting validation, too. Putting the validation on repository layer is not bulletproof (but maybe it is safer than service layer since repository layer has less modifications typically)
- I like @carohauta 's suggestion of putting the validation **logic** into somewhere else than the repository. I would still keep both validation **checks** (service and repository), just move the actual logic elsewhere. It would make the repository class a bit more consistent with other repository classes in what kind of **logic** it contains.
- repository level validation (if there is any, we should try to keep it limited) should be considered as a safety net like a database constraint is. Failing validation should be comparable to a programming error and can just throw some dirty RuntimeException, result in http 500, not care about nice error messages etc.

Reply:
Based on the comments, the validation logic is now moved to `FormatUtils` class, and both `BackupRepository` and `BackupService` use it from there. In addition, filename validation was added to all the methods in the repository class that read, create or delete backup files.